kie-api/src/main/java/org/kie/api/io/ResourceType.java

```diff
@@ -30,24 +30,28 @@
 
     private static final long serialVersionUID = 1613735834228581906L;
 
-    private String name;
+    private final String name;
 
-    private String description;
+    private final String description;
 
-    private String defaultExtension;
+    private final String defaultExtension;
 
-    private String[] otherExtensions;
+    private final String[] otherExtensions;
 
-    private String defaultPath;
+    private final String defaultPath;
+
+    private final boolean fullyCoveredByExecModel;
 
     private static final Map<String, ResourceType> CACHE = Collections.synchronizedMap(new HashMap<String, ResourceType>());
 
     public ResourceType(String name,
+                        boolean fullyCoveredByExecModel,
```

Comment:
@mariofusco 
Looking at how this is used inside the other PR (https://github.com/kiegroup/drools/pull/2926) I think this is not needed and misleading
1) it must have always the same value for a given type
2) there is always a direct relationship 1:1 between the type and this flag
3) it seems to only add level of indirectness (inside CanonicalKieModule.NON_MODEL_RESOURCES)